### PR TITLE
Drop support for --fetch/--no-fetch option for local scans

### LINKED ISSUE #203: tartufo crashes scanning local repo with no remote named origin
## 🐛 Bug Report

When running `tartufo scan-local-repo .` it crashes since I don't have a remote named `origin` (I don't use standard remote names for my repositories.)

```bash
❯ tartufo scan-local-repo .
Traceback (most recent call last):
  File "/usr/local/bin/tartufo", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/tartufo/commands/scan_local_repo.py", line 47, in main
    scanner.scan()
  File "/usr/local/lib/python3.9/site-packages/tartufo/scanner.py", line 342, in scan
    for chunk in self.chunks:
  File "/usr/local/lib/python3.9/site-packages/tartufo/scanner.py", line 592, in chunks
    self._repo.remotes.origin.fetch()
  File "/usr/local/lib/python3.9/site-packages/git/util.py", line 972, in __getattr__
    return list.__getattribute__(self, attr)
AttributeError: 'IterableList' object has no attribute 'origin'
```

## To Reproduce

1. clone a git repository and `cd` into the repo directory
2. rename the origin remote. i.e. `git remote rename origin foo`
3. run `tartufo scan-local-repo .`
4. observe error above ^

## Expected Behavior

1. `tartufo` **should not fail** regardless of what the repo's remote names are. 
2. Why does `scan-local-repo` inspect remotes at all?

## Code Example

N/A

## Environment

N/A

### LINKED ISSUE #240: Remove the `--fetch`/`--no-fetch` options in v3.0
## Is your feature request related to a problem? Please describe.

The `--fetch`/`--no-fetch` functionality is just weird. It only exists because it has existed since forking from `truffleHog`. It should no longer exist.

## Describe the solution you'd like

The options are completely gone!

## Describe alternatives you've considered

Switching the default to `--no-fetch`. But really I see no reason to leave it in at all. It should just go away. Users can easily `git fetch` before running tartufo, and then they have better control without weird side effects.

## Teachability, Documentation, Adoption, Migration Strategy

We may want to put a note into 2.x docs that this option is being deprecated and will be removed in v3.0.

### PR DESCRIPTION
**Pull request by sushantmimani**
To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [ ] Tests (if applicable)
* [ ] Documentation (if applicable)
* [x] Changelog entry
* [x] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [x] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

* [ ] Yes (backward compatible)
* [x] No (breaking changes)

## Issue Linking
resolves #240 

## What's new?
- This PR drops support for the --fetch/--no-fetch option for local scans.

### CODE REVIEW

File: tests/test_git_repo_scanner.py [hunk 1]
```diff
@@ -148,56 +148,11 @@ def test_filter_submodules_handles_broken_submodules_explicitly(
 
 
 class ChunkGeneratorTests(ScannerTestCase):
-    @mock.patch("pygit2.Repository")
-    def test_single_branch_is_loaded_if_specified(self, mock_repo: mock.MagicMock):
```
Comment (tarkatronic) [inline tests/test_git_repo_scanner.py:152]: This test is still necessary to make sure that specifying `--branch` limits the scan

**Review by tarkatronic — CHANGES_REQUESTED** [review]
Just need one test brought back in 😄

File: tests/test_git_repo_scanner.py (hunk 1 above)
Comment (sushantmimani) [reply tests/test_git_repo_scanner.py:152]: Updated & added back

**Review by tarkatronic — APPROVED** [review]

**Review by emayuri-godaddy — APPROVED** [review]
LGTM

**Review by rbailey-godaddy — APPROVED** [review]

**Pull request merged**